plugins/inputs/consul/consul.go

```diff
@@ -110,15 +127,21 @@ func (c *Consul) GatherHealthCheck(acc telegraf.Accumulator, checks []*api.Healt
 		record := make(map[string]interface{})
 		tags := make(map[string]string)
 
-		record["check_name"] = check.Name
-		record["service_id"] = check.ServiceID
-
-		record["status"] = check.Status
 		record["passing"] = 0
 		record["critical"] = 0
 		record["warning"] = 0
 		record[check.Status] = 1
 
+		if c.MetricVersion == 2 {
+			tags["check_name"] = check.Name
+			tags["service_id"] = check.ServiceID
+			tags["status"] = check.Status
```

Comment:
status seems like it should stay as a field?

Reply:
I think it should not. It will duplicate other fields in such case. Tag 'status' is just string representation of fields passing/critical/warning, it's value could be convenient to use in alert templates, for example. In my opinion it is more like metadata than value that one can add into some computations like get avg, max or min.